File: manual/data/java/tests/test_peewee_get.java
package com.example;

import org.junit.jupiter.api.*;
import static org.junit.jupiter.api.Assertions.*;

public class test_peewee_get {

    private String db_path;
    private function_peewee_get executor;

    @BeforeEach
    public void setUp() {
        // Use a shared database file for tests.
        db_path = "test_db.sqlite";
        executor = new function_peewee_get();
    }

    @Test
    public void test_existing_record() {
        assertEquals("Alice", executor.get_name_by_id(db_path, 1));
    }

    @Test
    public void test_existing_record_2() {
        assertEquals("Bob", executor.get_name_by_id(db_path, 2));
    }

    @Test
    public void test_nonexistent_record() {
        assertEquals("Not Found", executor.get_name_by_id(db_path, 99));
    }

    @Test
    public void test_zero_id() {
        assertEquals("Not Found", executor.get_name_by_id(db_path, 0));
    }

    @Test
    public void test_negative_id() {
        assertEquals("Not Found", executor.get_name_by_id(db_path, -1));
    }
}

File: manual/data/cpp/tests/test_peewee_get.cpp
#include <gtest/gtest.h>
#include "../src/function_peewee_get.cpp"


class TestPeeweeExecutor : public ::testing::Test {
protected:
    PeeweeExecutor executor;
    std::string db_path = "test_db.sqlite";

    void SetUp() override {
    }

    void TearDown() override {
    }
};

TEST_F(TestPeeweeExecutor, test_existing_record) {
    EXPECT_EQ(executor.get_name_by_id(db_path, 1), "Alice");
}

TEST_F(TestPeeweeExecutor, test_existing_record_2) {
    EXPECT_EQ(executor.get_name_by_id(db_path, 2), "Bob");
}

TEST_F(TestPeeweeExecutor, test_nonexistent_record) {
    EXPECT_EQ(executor.get_name_by_id(db_path, 99), "Not Found");
}

TEST_F(TestPeeweeExecutor, test_zero_id) {
    EXPECT_EQ(executor.get_name_by_id(db_path, 0), "Not Found");
}

TEST_F(TestPeeweeExecutor, test_negative_id) {
    EXPECT_EQ(executor.get_name_by_id(db_path, -1), "Not Found");
}

int main(int argc, char** argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}